hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/AlterHoodieTableAddColumnsCommand.scala

```diff
@@ -59,7 +64,7 @@ case class AlterHoodieTableAddColumnsCommand(
           s" table columns is: [${hoodieCatalogTable.tableSchemaWithoutMetaFields.fieldNames.mkString(",")}]")
       }
       // Get the new schema
-      val newSqlSchema = StructType(tableSchema.fields ++ colsToAdd)
+      val newSqlSchema = StructType(hoodieCatalogTable.dataSchema.fields ++ colsToAdd ++ hoodieCatalogTable.partitionSchema.fields)
```

Comment:
would you please clarify why need this change? must put partition schema in the end?